server/controllers/users.js

```diff
@@ -152,6 +169,91 @@ class UserController {
       next(error);
     }
   }
+
+  /**
+  * Sends mail to verify a new user
+  * @async
+  * @param  {object} req - Request object
+  * @param {object} res - Response object
+  * @param {object} next The next middleware
+  * @return {json} Returns json object
+  * @static
+  */
+  static async sendMailToVerifyAccount(req, res, next) {
+    try {
+      const {
+        active, email, id, username
+      } = req.user;
+
+      const { user } = req;
+      if (active === false) {
+        const verifyToken = randomstring.generate(40);
+        const tokenExpiry = Date.now() + ((Number(process.env.RESET_TOKEN_EXPIRE)) || 75600000);
+
+        const verifyDetails = {
+          verifyToken, tokenExpiry, userId: id
+        };
+
+        const userDetails = await user.getVerifiedUser({
+          where: {
+            userId: id
+          }
+        });
+
+        await userDetails
+          .update(verifyDetails);
+        sendVerifyMailToken(verifyToken, email, username);
+
+        return Response.success(res, 200, { verifyToken }, 'Verification mail sent');
+      }
+      if (active) {
+        return Response.error(res, 400, 'You are already verified');
+      }
+    } catch (err) {
+      next(err);
+    }
+  }
+
+  /**
+  * Verifies a new user
+  * @async
+  * @param  {object} req - Request object
+  * @param {object} res - Response object
+  * @param {object} next The next middleware
+  * @return {json} Returns json object
+  * @static
+  */
+  static async verifyAccount(req, res, next) {
+    try {
+      const { token, email } = req.query;
+
+      if (!token || !email) return Response.error(res, 400, 'Please use a valid reset link');
+      const user = await User.findOne({ where: { email } });
+      if (!user) return Response.error(res, 404, 'User does not exist');
```

Comment:
lol what?

firstly, this wouldn't exactly be a 404 (considering the username is not in the endpoint)
Also, you are releasing too much information by doing this.
`Invalid Token` would work here

Reply:
Alright on it.